### ISSUE TITLE
Trim dependencies: fastutil

### ISSUE THREAD

**Issue by lintool** [castorini/anserini#835]
We only use `fastutil` in two places:

```
    <dependency>
      <groupId>it.unimi.dsi</groupId>
      <artifactId>fastutil</artifactId>
      <version>6.5.6</version>
    </dependency>
```

If we're able to trim it out, it'd drop the fatjar size a fair bit, to around 84MB from around 100MB.

**Comment by lintool** [comment]
See rabbit's hole documented in #840

**Comment by lintool** [comment]
Closing this issue because the complete story is documented in #840, and #864 is a temporary fix for fastutil 6.5.6.
